playground/tests/nuxt/index.spec.ts

```diff
@@ -29,16 +30,24 @@ describe('client-side nuxt features', () => {
   it('defaults to index page', async () => {
     expect(useRoute().matched[0].meta).toMatchInlineSnapshot(`
       {
-        "slug": "foo",
+        "value": "set in index",
       }
     `)
     // TODO: should it be possible to push to other routes?
   })
 
-  it('allows pushing to other pages', async () => {
-    await useRouter().push('/something')
-    expect(useRoute().fullPath).toMatchInlineSnapshot('"/something"')
-  })
+  it('allows pushing to other pages', async () =>
+    new Promise<void>(done => {
+      useRouter()
+        .push('/something')
+        .then(() => {
+          const stop = watch(useRoute(), () => {
```

Comment:
Hoping this works 🤞 

Not a big fan of how the test reads (since Vitest deprecated the async callback form of test from Jest), but this approach to checking `useRoute` should at least be robust since we're using the reactive system from Vue.